sources/cfsignal/src/system_check.rs

```diff
@@ -0,0 +1,60 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct SystemHealth {
+    /// Whether or not the system is healthy.
+    pub(crate) is_healthy: bool,
+    /// The system status description.
+    pub(crate) status: String,
+    /// The system status code.
+    pub(crate) exit_code: i32,
+}
+
+pub(crate) struct SystemCheck {}
+
+impl SystemCheck {
+    pub fn system_running(&self) -> Result<SystemHealth> {
+        let mut result = is_system_running()?;
+        // remove trailing newline
+        result.stdout.pop();
+        Ok(SystemHealth {
+            is_healthy: result.is_exit_true(),
+            status: result.stdout,
+            exit_code: result.exit,
+        })
+    }
+}
+
+struct Outcome {
+    exit: i32,
+    stdout: String,
+}
+
+impl Outcome {
+    fn is_exit_true(&self) -> bool {
+        self.exit == 0
+    }
```

Comment:
nit: I'm wondering if this is more accurate.
```suggestion
    fn is_success(&self) -> bool {
        self.exit == 0
    }
```